File: CalculatorBackend/src/computer/MultiplyDivideProcessor.java
package computer;

import java.util.Arrays;

import ifc.IfcMathExpression;

public class MultiplyDivideProcessor implements IfcMathExpression {
    @Override
    public String compute(String term) {
    	
    	//term = term.replace("+-", "-").replace("--", "+"); //So etwas muss man nicht machen mit Punktoperatoren.
    	
        if (!(term.contains("/") || term.contains("*"))) {

        	return term;
        
        }

        //String regex = "(?<=[+*/])-|(?=[+*/])|(?<=\\d)-(?=\\d)|(?<=\\()|(?=\\))";
        //String[] numbers = term.split(regex);
        String[] numbers = term.split("[+\\-*/]");
        
        numbers = Arrays.stream(numbers)
                .filter(s -> !s.isEmpty()) // Entfernt leere Strings
                .toArray(String[]::new);

        
        String[] operators = term.replaceAll("[0-9.]", "").split("");

        for (int i = 0; i < operators.length; i++) {
            if (operators[i].equals("/") || operators[i].equals("*")) {
                String tmpResult = calculate(numbers[i], operators[i], numbers[i + 1]);
                String newTerm = rebuildTerm(numbers, operators, i, tmpResult);
                return compute(newTerm); // Rekursion für verbleibende * und /
            }
        }

        return term;
    }

    private String calculate(String num1Str, String operator, String num2Str) {
        double num1 = Double.parseDouble(num1Str);
        double num2 = Double.parseDouble(num2Str);
        double result = operator.equals("/") ? num1 / num2 : num1 * num2;
        return Double.toString(result);
    }

    private String rebuildTerm(String[] numbers, String[] operators, int index, String result) {
        StringBuilder newTerm = new StringBuilder();
        for (int i = 0; i < numbers.length; i++) {
            if (i == index) {
                newTerm.append(result);
                i++;
            } else {
                newTerm.append(numbers[i]);
            }
            if (i < operators.length && i != index) {
                newTerm.append(operators[i]);
            }
        }
        return newTerm.toString();
    }
}

File: computer/MultiplyDivideProcessor.cs
using CalculatorBackend.ifc;

public class MultiplyDivideProcessor : IfcMathExpression
{
    public string Compute(string term)
    {
        if (!(term.Contains("/") || term.Contains("*")))
        {
            return term;
        }

        string[] numbers = term.Split(new char[] { '+', '-', '*', '/' }, StringSplitOptions.RemoveEmptyEntries);
        string[] operators = term.Where(c => "+-*/".Contains(c)).Select(c => c.ToString()).ToArray();

        for (int i = 0; i < operators.Length; i++)
        {
            if (operators[i] == "/" || operators[i] == "*")
            {
                string tmpResult = Calculate(numbers[i], operators[i], numbers[i + 1]);
                string newTerm = RebuildTerm(numbers, operators, i, tmpResult);
                return Compute(newTerm); // Rekursion für verbleibende * und /
            }
        }

        return term;
    }

    private string Calculate(string num1Str, string operatorStr, string num2Str)
    {
        double num1 = double.Parse(num1Str);
        double num2 = double.Parse(num2Str);
        double result = operatorStr == "/" ? num1 / num2 : num1 * num2;
        return result.ToString();
    }

    private string RebuildTerm(string[] numbers, string[] operators, int index, string result)
    {
        var newTerm = "";
        for (int i = 0; i < numbers.Length; i++)
        {
            if (i == index)
            {
                newTerm += result;
                i++;
            }
            else
            {
                newTerm += numbers[i];
            }
            if (i < operators.Length && i != index)
            {
                newTerm += operators[i];
            }
        }
        return newTerm;
    }
}
